staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -456,6 +457,10 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 			isSync := d.Type == Sync
 			s.cacheMutationDetector.AddObject(d.Object)
 			if old, exists, err := s.indexer.Get(d.Object); err == nil && exists {
+				// If the delta is from a relist, and the RV hasn't changed, elide it.
```

Comment:
We also compare UIDs. Comment can be updated to reflect the behavior of `isObjectSame`

Reply:
removed the UID comparison.